src/frontends/lean/inductive_cmds.cpp

```diff
@@ -693,6 +697,61 @@ class inductive_cmd_fn {
         parse_mutual_inductive(params, inds, intro_rules);
         return shared_inductive_cmd(params, inds, intro_rules);
     }
+
+    environment shared_coinductive_cmd(buffer<expr> const & params,
+                                       buffer<expr> const & inds,
+                                       buffer<buffer<expr> > const & intro_rules) {
+        buffer<expr> new_params;
+        buffer<expr> new_inds;
+        buffer<buffer<expr> > new_intro_rules;
+        if (!m_explicit_levels) {
+            m_explicit_levels = has_explicit_level(intro_rules);
+        }
+        elaborate_inductive_decls(params, inds, intro_rules, new_params, new_inds, new_intro_rules);
+
+        expr first_ind = inds[0];
+        metavar_context mctx;
+        options opts = m_p.get_options();
+
+        vm_state vms(m_env, opts);
+        scope_vm_state vms_scope(vms);
+        tactic_state s = mk_tactic_state_for(m_env, opts, mlocal_name(first_ind), mctx, local_context(), mk_true());
```

Comment:
I just pushed a relevant fix: 
https://github.com/leanprover/lean/commit/544817cf150a726dbc531e7eb816f1921c478868
The lines 
```lean
  vm_state vms(m_env, opts);
  scope_vm_state vms_scope(vms);
```
are not needed anymore. The `tactic_evaluator` creates its own VM.